src/awst_build/eb/bytes-expression-builder.ts

```diff
@@ -273,6 +279,63 @@ export class BytesInvertBuilder extends ParameterlessFunctionBuilder {
   }
 }
 
+export class BitwiseAndExpressionBuilder extends FunctionBuilder {
+  constructor(private expr: awst.Expression) {
+    super(expr.sourceLocation)
+  }
+
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const [other] = requireExpressionsOfType(args, [bytesPType], sourceLocation)
```

Comment:
This should make use of `parseFunctionArgs`. It handles checking generic types and produces better error messages. The code would look something like this. Here is an example of the `ConcatExpressionBuilder` refactored to use it.

```ts
  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
    const {
      args: [other],
    } = parseFunctionArgs({
      args,
      typeArgs,
      genericTypeArgs: 0,
      callLocation: sourceLocation,
      funcName: 'Concat',
      argSpec: (a) => [a.required(bytesPType)],
    })
    return new BytesExpressionBuilder(
      intrinsicFactory.bytesConcat({
        left: this.expr,
        right: other.resolve(),
        sourceLocation: sourceLocation,
      }),
    )
  }
```

`ConcatExpressionBuilder`, and your introduced builder should also use the naming convention of `XxxxFunctionBuilder` rather than Expression builder. This is obviously existing tech debt but may as well clean it up whilst you're there :)

Reply:
It has been refactored. f369afc